tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -2992,6 +2999,102 @@ TEST_F(DataserviceReadVersionedLayerClientTest, CheckLookupApiCacheExpiration) {
+    constexpr auto begin =
+        R"jsonString({ "partitions": [{"version":4,"partition":")jsonString";
+    constexpr auto middle =
+        R"jsonString(","layer":"testlayer","dataHandle":")jsonString";
+    constexpr auto end = R"jsonString("}]})jsonString";
+    return begin + partition + middle + data_handle + end;
+  };
+
+  auto is_in_cache = [&client](const std::string& key) {
+    const auto request = olp::dataservice::read::DataRequest()
+                             .WithPartitionId(key)
+                             .WithFetchOption(FetchOptions::CacheOnly);
+    auto future = client->GetData(request).GetFuture();
+    return future.get().IsSuccessful();
+  };
+
+  ON_CALL(*network_mock_, Send(IsGetRequest(URL_LOOKUP_QUERY), _, _, _, _))
+      .WillByDefault(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                            olp::http::HttpStatusCode::OK),
+                                        kHttpResponseLookupQuery));
+  ON_CALL(*network_mock_, Send(IsGetRequest(URL_LOOKUP_BLOB), _, _, _, _))
+      .WillByDefault(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                            olp::http::HttpStatusCode::OK),
+                                        kHttpResponseLookupBlob));
+
+  const auto promote_key = std::to_string(0);
+  const auto evicted_key = std::to_string(1);
+
+  // overflow the mutable cache
+  auto count = 0u;
+  auto partition = std::to_string(count);
+  const auto max_count = cache_settings.max_disk_storage / data_size;
+  for (; count < max_count; ++count) {
+    partition = std::to_string(count);
+    const auto data_handle = data_handle_prefix + partition;
+    const auto partition_response =
+        create_partition_response(partition, data_handle);
+
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequestPrefix(kUrlPartitionsPrefix), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     partition_response));
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequestPrefix(kUrlBlobstorePrefix), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     blob_data));
+
+    const auto request =
+        olp::dataservice::read::DataRequest().WithPartitionId(partition);
+    auto future = client->GetData(request).GetFuture();
+
+    ASSERT_NE(future.wait_for(kWaitTimeout), std::future_status::timeout);
+    ASSERT_TRUE(future.get().IsSuccessful());
+
+    // promote the key so its not evicted.
+    ASSERT_TRUE(is_in_cache(promote_key));
+  }
+
+  // maximum is reached.
+  ASSERT_TRUE(count == max_count);
+  EXPECT_TRUE(is_in_cache(promote_key));
+  EXPECT_TRUE(is_in_cache(promote_key));
```

Comment:
Duplicate?

Reply:
Oops, it should test last added element is cached. Fixed!